lib/internal/fs/watch/linux.js

```diff
@@ -0,0 +1,244 @@
+'use strict';
+
+const { SafeMap, Symbol, StringPrototypeStartsWith, SymbolAsyncIterator, Promise } = primordials;
+
+const {
+  ERR_FEATURE_UNAVAILABLE_ON_PLATFORM,
+  codes: { ERR_INVALID_ARG_TYPE, ERR_INVALID_ARG_VALUE },
+} = require('internal/errors');
+const { kEmptyObject } = require('internal/util');
+const { validateString, validateObject, validateBoolean, validateAbortSignal } = require('internal/validators');
+const { EventEmitter } = require('events');
+const path = require('path');
+
+const kFSWatchStart = Symbol('kFSWatchStart');
+
+let internalSync;
```

Comment:
Can we get rid of this and instead use `const { watch } = require('internal/fs/watchers');`?

Reply:
That is the native implementation, which Windows and OSX relies on. I did not understand your recommendation.